module/mempool/stdmap/pending_receipts.go

```diff
@@ -0,0 +1,109 @@
+package stdmap
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// PendingReceipts stores pending receipts indexed by the id.
+// It also maintains a secondary index on the previous result id, which is unique,
+// in order to allow to find a receipt by the previous result id.
+type PendingReceipts struct {
+	*Backend
+	// secondary index by parent result id, has to be unqiue
+	byPreviousResultID map[flow.Identifier]flow.Identifier
+}
+
+func indexByPreviousResultID(receipt *flow.ExecutionReceipt) flow.Identifier {
+	return receipt.ExecutionResult.PreviousResultID
+}
```

Comment:
To me, it would be clearer to work with `receipt.ExecutionResult.PreviousResultID` directly instead of wrapping it into a function.

Reply:
it's more to specify which field is the secondary index, it's a way to guarantee everyone uses the same index. 

In addition, we could potentially go further to extract the common code of building a secondary index, so the only thing you need to provide, is a function to specify which field (or fields) is the secondary index, which is this `indexByPreviousResultID` fucntion, and the rest are reused code.